src/main/java/org/openrewrite/staticanalysis/ReplaceCollectToListWithToList.java

```diff
@@ -0,0 +1,92 @@
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.J;
+
+import java.time.Duration;
+
+/**
+ * Replaces Stream.collect(Collectors.toList()) with Stream.toList()
+ * <p>
+ * <a href="https://github.com/apache/maven/pull/2322">[POC] rewrite-maven-plugin: Introduce OpenRewrite by Moderne</a>
+ * <a href="https://github.com/apache/maven/pull/2277">Modernize codebase with Java improvements</a>
+ * <a href="https://github.com/apache/maven/pull/2277#pullrequestreview-2814419390">suppression request</a>
+ *
+ */
+public class ReplaceCollectToListWithToList extends Recipe {
+
+    private static final MethodMatcher COLLECT_TO_LIST = new MethodMatcher("java.util.stream.Stream collect(java.util.stream.Collector)");
+
+    @Override
+    public String getDisplayName() {
+        return "Replace stream.collect(collectors.toList()) with stream.toList()";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace Java 11 `Stream.collect(Collectors.toList())` with Java 16+ `Stream.toList()` for more concise syntax.";
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(1);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(Preconditions.and(
+                        new UsesJavaVersion<>(16),
+                        new UsesMethod<>(COLLECT_TO_LIST)),
+                new JavaVisitor<ExecutionContext>() {
+
+                    private final JavaTemplate template = JavaTemplate
+                            .builder("#{any(java.util.stream.Stream)}.toList()")
+                            .build();
+
+                    @Override
+                    public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                        J.MethodInvocation m = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
+
+                        if (COLLECT_TO_LIST.matches(m) && isCollectorsToList(m)) {
+                            maybeRemoveImport("java.util.stream.Collectors");
+                            return template.apply(
+                                    updateCursor(m),
+                                    m.getCoordinates().replace(),
+                                    m.getSelect()
+                            );
+                        }
+                        return m;
+                    }
+
+                    private boolean isCollectorsToList(J.MethodInvocation method) {
+                        return method.getArguments().get(0) instanceof J.MethodInvocation &&
+                                ((J.MethodInvocation) method.getArguments().get(0)).getSimpleName().equals("toList") &&
+                                ((J.MethodInvocation) method.getArguments().get(0)).getSelect() != null &&
+                                ((J.MethodInvocation) method.getArguments().get(0)).getSelect().toString().equals("Collectors");
```

Comment:
This would fail to pick up static imports of `toList`. Wherever possible prefer to use MethodMatchers as opposed to looking at simple names, as that avoids these pitfalls.